ArduCopter/avoidance_adsb.cpp

```diff
@@ -20,15 +20,15 @@ MAV_COLLISION_ACTION AP_Avoidance_Copter::handle_avoidance(const AP_Avoidance::O
         copter.failsafe.adsb = true;
         failsafe_state_change = true;
         // record flight mode in case it's required for the recovery
-        prev_control_mode = copter.control_mode;
+        prev_control_mode = copter.flightmode->mode_number();
     }
 
     // take no action in some flight modes
-    if (copter.control_mode == Mode::Number::LAND ||
```

Comment:
This is also not your bug, but this should be asking if the vehicle is landing for any reason (`is_landing`) - we could be doing an auto-mode landing and we ought not stop doing that.

Perhaps a future PR...